### PR TITLE
melgeek001365 [Keyboard] Update MJ61 architecture to support multi version

### PR DESCRIPTION
**Pull request by melgeek001365**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/melgeek/mj61/rev1/rev1.c [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A2);
+    writePinHigh(A2);
+    keyboard_pre_init_user();
+}
+#endif
```
Comment (drashna) [inline keyboards/melgeek/mj61/rev1/rev1.c:124]:
```suggestion
#endif

void keyboard_pre_init_kb(void) {
    setPinOutput(A2);
    writePinHigh(A2);
    keyboard_pre_init_user();
}
```

File: keyboards/melgeek/mj61/rev2/rev2.c [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (drashna) [inline keyboards/melgeek/mj61/rev2/rev2.c:97]:
```suggestion
```

File: keyboards/melgeek/mj61/rev2/rev2.c [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A2);
+    writePinHigh(A2);
+    keyboard_pre_init_user();
+}
+#endif
```
Comment (drashna) [inline keyboards/melgeek/mj61/rev2/rev2.c:211]:
```suggestion
#endif

void keyboard_pre_init_kb(void) {
    setPinOutput(A2);
    writePinHigh(A2);
    keyboard_pre_init_user();
}
```

File: keyboards/melgeek/mj61/rev1/rev1.c [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		{ 52,     NO_LED,    51,       50,       49,       48,       47,       46,       45,         44,        43,         42,       41,        NO_LED }, /* R3 */
+		{ 53,     54,        NO_LED,   55,       61,       56,       62,   NO_LED,   NO_LED,         57,        58,         59,       60,        NO_LED }, /* R4 */
+    }, {
+        {0, 0},   {16, 0},   {32, 0},  {48, 0},  {64, 0},  {80, 0},  {96, 0},  {112, 0},  {128, 0},  {144, 0},  {160, 0},  {176, 0},  {192, 0},  {208, 0},
+		{0, 16},  {16, 16},  {32, 16}, {48, 16}, {64, 16}, {80, 16}, {96, 16}, {112, 16}, {128, 16}, {144, 16}, {160, 16}, {176, 16}, {192, 16}, {208, 16},
+		{0, 32},  {16, 32},  {32, 32}, {48, 32}, {64, 32}, {80, 32}, {96, 32}, {112, 32}, {128, 32}, {144, 32}, {160, 32}, {176, 32}, {192, 32},
+		{0, 48},             {16, 48}, {32, 48}, {48, 48}, {64, 48}, {80, 48}, {96, 48},  {112, 48}, {128, 48}, {144, 48}, {160, 48}, {176, 48},
+		{0, 64},  {16, 64},            {32, 64}, {48, 64}, {64, 64}, {80, 64},                       {96, 64},  {112, 64}, {128, 64}, {144, 64},
+	}, {
```
Comment (drashna) [inline keyboards/melgeek/mj61/rev1/rev1.c:100]: Wait, so you have no modifiers on your keyboard, only alphas?

File: keyboards/melgeek/mj61/rev2/rev2.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+		{ 61,     62,        NO_LED,   63,       NO_LED,   65,       NO_LED,   NO_LED,   NO_LED,     67,        68,         69,       NO_LED,    70     }, /* R4 */
+    }, {
+        {0, 0},   {16, 0},   {32, 0},  {48, 0},  {64, 0},  {80, 0},  {96, 0},  {112, 0},  {128, 0},  {144, 0},  {160, 0},  {176, 0},  {192, 0},  {208, 0},  {224, 0},  {240, 0},
+		{0, 16},  {16, 16},  {32, 16}, {48, 16}, {64, 16}, {80, 16}, {96, 16}, {112, 16}, {128, 16}, {144, 16}, {160, 16}, {176, 16}, {192, 16}, {208, 16},
+		{0, 32},  {16, 32},  {32, 32}, {48, 32}, {64, 32}, {80, 32}, {96, 32}, {112, 32}, {128, 32}, {144, 32}, {160, 32}, {176, 32}, {192, 32}, {208, 32}, {224, 32},
+		{0, 48},  {16, 48},  {32, 48}, {48, 48}, {64, 48}, {80, 48}, {96, 48}, {112, 48}, {128, 48}, {144, 48}, {160, 48}, {176, 48}, {192, 48}, {208, 48}, {224, 48}, {240, 48},
+		{0, 64},  {16, 64},  {32, 64}, {48, 64}, {64, 64}, {80, 64}, {96, 64}, {112, 64}, {128, 64}, {144, 64},
+	}, {
+	       4,      4,         4,        4,        4,        4,        4,        4,         4,         4,         4,          4,        4,        4,         4,         4,
```
Comment (drashna) [inline keyboards/melgeek/mj61/rev2/rev2.c:188]: All alphas, no modifiers?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/mj61/rev1/rev1.c (hunk 4 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj61/rev1/rev1.c:100]: yes. all of leds for key backlight

File: keyboards/melgeek/mj61/rev2/rev2.c (hunk 5 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj61/rev2/rev2.c:111]: in fact, I don't very clear about modifer flag of led

File: keyboards/melgeek/mj61/rev1/rev1.c (hunk 4 above)
Comment (drashna) [reply keyboards/melgeek/mj61/rev1/rev1.c:100]: So no escape/tab/capslock/shift/ctrl/ etc?

File: keyboards/melgeek/mj61/rev2/rev2.c (hunk 5 above)
Comment (drashna) [reply keyboards/melgeek/mj61/rev2/rev2.c:111]: The flags are used by a number of animations.  
https://docs.qmk.fm/#/feature_rgb_matrix?id=flags

So you can have alphas (keylight), modifiers (esc, tab, caps, shift, control, etc), underglow, and indicators.   

Additionally, people (like myself) use these flags for layer indication and the like.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**